src/backend/app.js

```diff
@@ -20,11 +20,74 @@ app.use(express.json());
 app.use(cors());
 
 router.use("/meals", mealsRouter);
+app.get("/my-route", (req, res) => {
+  res.send("Hi friend");
+});
+//future-meals
+app.get("/future-meals", async (req, res) => {
+  try {
+    const [rows] = await knex.raw("SELECT * FROM meal WHERE YEAR(when) = 2022");
+    res.json(rows);
+  } catch (e) {
+    console.error(e);
+    res.status(500).json({ error: "Internal server error" });
+  }
+});
+//past-meals
+app.get("/past-meals", async (req, res) => {
+  try {
+    const [rows] = await knex.raw("SELECT * FROM meal WHERE YEAR(when) = 2021");
```

Comment:
it would be convenient to implement @itsgauravsaxena suggestion, since `< NOW() or > NOW ` compares past dates and future dates respectively not only years (as you implemented) but it compares dates too.